src/frontend/screens/ManualGpsScreen/DdForm.tsx

```diff
@@ -0,0 +1,250 @@
+    defaultMessage: "West",
+  },
+});
+
+const DdForm = ({ location, onValueUpdate }: FormProps) => {
+  const { formatMessage: t } = useIntl();
+
+  const DIRECTION_OPTIONS_NORTH_SOUTH = [
+    {
+      value: "N",
+      label: t(m.north),
+    },
+    {
+      value: "S",
+      label: t(m.south),
+    },
+  ];
+
+  const DIRECTION_OPTIONS_EAST_WEST = [
+    {
+      value: "E",
+      label: t(m.east),
+    },
+    {
+      value: "W",
+      label: t(m.west),
+    },
+  ];
+
+  const [latidudeDegrees, setLatitudeDegrees] = React.useState<string>(() => {
+    if (!location.savedPosition) return "";
+
+    const { latitude } = location.savedPosition.coords;
+    return Math.abs(latitude).toString();
+  });
+  const [longitudeDegrees, setLongitudeDegrees] = React.useState<string>(() => {
+    if (!location.savedPosition) return "";
+
+    const { longitude } = location.savedPosition.coords;
+    return Math.abs(longitude).toString();
+  });
+
+  const [selectedLatCardinality, setSelectedLatCardinality] = React.useState(
+    getInitialCardinality("lat", location)
+  );
+  const [selectedLonCardinality, setSelectedLonCardinality] = React.useState(
+    getInitialCardinality("lon", location)
+  );
+
+  const parsedLat = parseNumber(latidudeDegrees);
+  const parsedLon = parseNumber(longitudeDegrees);
+
+  const signedLat =
+    parsedLat !== undefined
+      ? parsedLat * (selectedLatCardinality === "N" ? 1 : -1)
+      : undefined;
+  const signedLon =
+    parsedLon !== undefined
+      ? parsedLon * (selectedLonCardinality === "E" ? 1 : -1)
+      : undefined;
+
+  React.useEffect(() => {
+    try {
+      const latIsValidRange =
+        signedLat !== undefined && Math.abs(signedLat) <= 90;
+      const lonIsValidRange =
+        signedLon !== undefined && Math.abs(signedLon) <= 180;
+
+      if (latIsValidRange && lonIsValidRange) {
+        onValueUpdate({
+          coords: {
+            lat: signedLat,
+            lon: signedLon,
+          },
+        });
+      } else {
+        throw new Error(m.invalidCoordinates.defaultMessage);
+      }
+    } catch (err) {
+      onValueUpdate({
+        error: err,
+      });
+    }
+  }, [signedLat, signedLon, onValueUpdate]);
+
+  const validateOnChange = (setState: (v: string) => void) => (
+    text: string
+  ) => {
+    const isJustDecimal = text === ".";
+    const endsWithDecimal = text.endsWith(".");
+
+    if (
+      text.length === 0 ||
+      isJustDecimal ||
+      endsWithDecimal ||
+      POSITIVE_DECIMAL_REGEX.test(text)
+    ) {
+      setState(text);
+    } else if (Number.parseFloat(text) === 0) {
+      setState("0");
+    }
+  };
+
+  const formatInputValue = (setState: (v: string) => void) => ({
+    nativeEvent: { text },
+  }: NativeSyntheticEvent<TextInputEndEditingEventData>) => {
+    const parsed = Number.parseFloat(text);
+    if (!isNaN(parsed)) {
+      // Do we want to enforce precision here?
+      setState(parsed.toFixed(6));
```

Comment:
highlighting this. we we stop editing the field in the DD format, we format it with 6 decimal places. not sure if that's too granular or if it's okay

Reply:
> behaviour of formatting to fewer decimal places when changing field

I actually enforce a maximum length for text inputs here (11 characters, which includes a potential decimal), so the behavior you'd more likely see is that we'd _add_ decimal places when changing fields e.g. inputting `1.50` gets formatted to `1.500000`, which is still a little weird so i may just omit the `toFixed` anyways